Apply minor fixes to kx

sskeirik opened the pull request:
kx command `-o` option is broken when not `pretty` or `kore` because of incorrect `kast` command invocation.

yilongli commented:
@dwightguth please review

Commit:
kx: remove extra whitespace

Commit:
kx: fix kast command

dwightguth reviewed (approved):
This looks good in terms of the changes itself, but it looks like this isn't the only place where kast is called in this script without a -d and you may also want to fix those.

Commit:
kx: fix/simplify all kast invocations

Commit:
kx: check if *-kompiled directory exists

NB: this fix guards against path expansion failure when searching for
the *-kompiled directory.

Commit:
kx: fix parser command invocation

NB: using bash arrays ensures that single arguments with spaces, e.g.
directory names, do not get split when invoking the parser.

sskeirik committed:
Merge branch 'master' into kx_fix
